app/scripts/lib/createDupeReqFilterMiddleware.ts

```diff
@@ -0,0 +1,86 @@
+import { JsonRpcMiddleware } from 'json-rpc-engine';
+import log from 'loglevel';
+
+const SECOND = 1000;
+const MINUTE = SECOND * 60;
+const FIVE_MINUTES = MINUTE * 5;
+
+/**
+ * Creates a set abstraction whose values expire after five minutes.
+ *
+ * @returns The expiry set.
+ */
+const makeExpirySet = () => {
+  const seenValues: (number | string)[] = [];
+  const timestamps: number[] = [];
+
+  setTimeout(() => {
+    if (timestamps.length === 0) {
+      return;
+    }
+
+    const cutoffTime = Date.now() - FIVE_MINUTES;
```

Comment:
I picked five minutes because it seems extremely unlikely that we'd receive an actually duplicate request id in that timespan. Curious if we could go with an even shorter timespan, but I would need guidance from someone more familiar with MV3.